src/TEdit/View/WorldRenderXna.xaml.cs

```diff
@@ -1028,15 +1043,31 @@ private void DrawTileWalls(bool drawInverted = false)
                     var curtile = _wvm.CurrentWorld.Tiles[x, y];
                     if ((curtile.WallColor == 30) != drawInverted) continue;
 
+                    // Hide all walls not within a filter when enabled.
+                    bool forceGrayscale = false;
+                    if (FilterManager.WallIsNotAllowed(curtile.Wall))
+                        if (FilterManager.CurrentFilterMode == FilterManager.FilterMode.Hide) continue;
+                        else if (FilterManager.CurrentFilterMode == FilterManager.FilterMode.Grayscale) forceGrayscale = true;
+
+                    /// <summary>
+                    /// Returns the tile at (nx, ny) if within bounds; otherwise returns null.
+                    /// Use this for wall neighbor lookups (no filter checks).
+                    /// </summary>
+                    Tile GetWallNeighbor(int nx, int ny)
+                    {
+                        if (nx < 0 || ny < 0 || nx >= width || ny >= height) return null;
+                        return _wvm.CurrentWorld.Tiles[nx, ny];
+                    }
+
                     //Neighbor tiles are often used when dynamically determining which UV position to render
-                    neighborTile[e] = (x + 1) < width ? _wvm.CurrentWorld.Tiles[x + 1, y] : null;
-                    neighborTile[n] = (y - 1) > 0 ? _wvm.CurrentWorld.Tiles[x, y - 1] : null;
-                    neighborTile[w] = (x - 1) > 0 ? _wvm.CurrentWorld.Tiles[x - 1, y] : null;
-                    neighborTile[s] = (y + 1) < height ? _wvm.CurrentWorld.Tiles[x, y + 1] : null;
-                    neighborTile[ne] = (x + 1) < width && (y - 1) > 0 ? _wvm.CurrentWorld.Tiles[x + 1, y - 1] : null;
-                    neighborTile[nw] = (x - 1) > 0 && (y - 1) > 0 ? _wvm.CurrentWorld.Tiles[x - 1, y - 1] : null;
-                    neighborTile[sw] = (x - 1) > 0 && (y + 1) < height ? _wvm.CurrentWorld.Tiles[x - 1, y + 1] : null;
-                    neighborTile[se] = (x + 1) < width && (y + 1) < height ? _wvm.CurrentWorld.Tiles[x + 1, y + 1] : null;
+                    neighborTile[e] = GetWallNeighbor(x + 1, y);

```

Comment:
If you remove the in-lining it adds a lot of overhead. 

`[MethodImpl(MethodImplOptions.AggressiveInlining)]` _might_ offset some of this, but it isn't perfect